### PR TITLE
Add layered_view visualization for model architectures

### LINKED ISSUE #21421: [Feature Suggestion] Improving Built-In Visualization System With Layered-Style Architecture Generation
I've been playing around with Keras's built-in visualizations tool for a while and I feel like it could be improved a lot. The current visualization system doesn't provide much information "at a glance" (meaning it takes quite a bit of analysis to understand what's going on) and isn't super aesthetically pleasing.

I'm opening this issue to see if an improved visualization system is something that the Keras team would be open to. I'd be happy to make small improvements to the existing system, but I feel like it would be better to add a new visualization system (or submodule) for highly customizable visual architecture generation. I could bring some (or most) of the key features from [visualkeras](https://github.com/paulgavrikov/visualkeras), a repository I maintain, into the `src/visualization` module.

Since we'd isolate the dependencies of the new features from Keras as a whole (like what was done for the last system), we wouldn't be adding any bloat through more dependencies.

Below are a few examples of what the visualizations I'd add may look like

Image

Image

Image

Image

Image

### PR DESCRIPTION
**Pull request by timon0305**
## Summary
- Adds `keras.visualization.layered_view()` for rendering model architectures as 3D block diagrams
- Each layer is drawn as a scaled 3D block reflecting its output shape
- Supports customizable colors, 2D/3D modes, funnel connectors, text labels, and legends
- Uses Pillow only (no Graphviz/PyDot dependency needed)
- Resolves #21421

### CODE REVIEW

**Comment by hertschuh** [comment]
This seems really cool. Can you attach some screenshots?

**Comment by timon0305** [comment]
## Demo Screenshots

### Simple CNN — 3D with funnel connectors
layered_view_3d_funnel

### Simple CNN — 3D without funnel
layered_view_3d_no_funnel


### Simple CNN — 2D flat view
layered_view_2d

### VGG-like deep model — 3D with funnel
layered_view_vgg

---

Generated with:
```python
model = keras.Sequential([
    keras.layers.Input(shape=(28, 28, 1)),
    keras.layers.Conv2D(32, (3, 3), activation='relu'),
    keras.layers.MaxPooling2D((2, 2)),
    keras.layers.Conv2D(64, (3, 3), activation='relu'),
    keras.layers.MaxPooling2D((2, 2)),
    keras.layers.Flatten(),
    keras.layers.Dense(64, activation='relu'),
    keras.layers.Dense(10, activation='softmax'),
])
keras.visualization.layered_view(model, legend=True)
```

File: keras/src/visualization/layered_view_test.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    def test_unbuilt_model_raises(self):
+        model = models.Sequential([layers.Dense(10)])
+        with self.assertRaises(ValueError):
+            layered_view(model)
+
+    def test_api_access(self):
+        import keras
+
+        self.assertTrue(hasattr(keras.visualization, "layered_view"))
```
Comment (hertschuh) [inline keras/src/visualization/layered_view_test.py:93]: Remove this, we don't do this.

File: keras/src/visualization/layered_view_test.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        img = layered_view(
+            model,
+            text_callable=lambda layer: layer.__class__.__name__,
+        )
+        self.assertIsNotNone(img)
+
+    def test_save_to_file(self):
+        model = self._build_sequential_cnn()
+        with tempfile.TemporaryDirectory() as tmpdir:
+            filepath = os.path.join(tmpdir, "model.png")
```
Comment (hertschuh) [inline keras/src/visualization/layered_view_test.py:80]: nitpick: we have a utility that auto cleans up and allows you to not have indented code.

```python
filepath = os.path.join(self.get_temp_dir(), "model.png")
```

File: keras/src/visualization/layered_view.py [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+from keras.src.api_export import keras_export
+
+try:
+    from PIL import Image
+    from PIL import ImageDraw
+    from PIL import ImageFont
+except ImportError:
+    Image = None
+    ImageDraw = None
+    ImageFont = None
```
Comment (hertschuh) [inline keras/src/visualization/layered_view.py:33]: We have a mechanism to make this cleaner:
- at the end of this file https://github.com/keras-team/keras/blob/master/keras/src/utils/module_utils.py you add `PIL = LazyModule("PIL", "pillow")
- here you simply do `from from keras.src.utils.module_utils import PIL`
- then you use `PIL.Image`, `PIL.ImageDraw` etc. in the code

File: keras/src/visualization/layered_view.py [hunk 4]
```diff
@@ -0,0 +1,428 @@
...
+}
+_FALLBACK_COLOR = (80, 80, 80, 255)
+
+
+def _check_pillow():
+    if Image is None:
+        raise ImportError(
+            "You must install Pillow (`pip install Pillow`) "
+            "for `layered_view` to work."
+        )
```
Comment (hertschuh) [inline keras/src/visualization/layered_view.py:69]: Remove this. The LazyModule system will throw this error with this message automatically when accessed.

File: keras/src/visualization/layered_view.py [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+        spacing: Horizontal spacing (pixels) between layers.
+        padding: Padding (pixels) around the image.
+        background_fill: Background color (color name or tuple).
+        text_callable: Optional callable ``f(layer) -> str`` to
+            generate text labels drawn above each layer block.
+        legend: If ``True``, append a color legend to the image.
+
+    Returns:
+        A ``PIL.Image.Image`` instance. Also saved to ``to_file``
+        if provided.
```
Comment (hertschuh) [inline keras/src/visualization/layered_view.py:297]: Change all double backticks to single backticks in this file, otherwise they won't render correctly on keras.io
